### DISCUSSION TITLE
proTable 中 request 得到的資料不會顯示出來，拜託大家了

### DISCUSSION THREAD

**Discussion by sam0324sam** [ant-design/ant-design#40621] [Q&A] [answered]
去年開發的後台，在重新架設環境後遇到的問題
使用antd 3的版本
程式碼部分在去年是正常的
結果這次就出問題了，抓到的資料不會顯示
![image](https://user-images.githubusercontent.com/9580348/217388073-680f47b2-7fbe-4e12-beed-87121c44e70d.png)
![image](https://user-images.githubusercontent.com/9580348/217388209-262c8d66-358a-42bf-8b1c-6a4a9535002d.png)
![image](https://user-images.githubusercontent.com/9580348/217388565-2cda9b61-eb67-406f-9170-97d478e59c63.png)

log上都有顯示我抓到的資料，所以API都沒問題
但是網頁上表格就一直都是一片空白QQ
拜託各位大佬了
這問題已經找好幾天了

**Comment by yoyo837** [comment]
1.尝试写死一个带内容数组直接return；
2.既然使用await，为什么不const arr = await而要在then里给外部变量赋值？

**Reply by sam0324sam** [reply]
這點我會再研究看看..是第一次寫，很多地方還在摸索

**Comment by hcisme** [comment, marked as answer]
你的ProTable里面的reuqest必须返回一个 Promise对象 对象中必须包括  `{ data: [ ] success: true, total: 0 }`，如果需要手动分页 total 也是必需的    [文档](https://procomponents.ant.design/components/table?current=1&pageSize=5#request)

**Reply by sam0324sam** [reply]
有了!
太久沒用都忘了要有success跟total了
不小心return到下一層了
